### PR TITLE
[Docs] update `addXYZ` functions

### LINKED ISSUE #2273: Document `addXYZ` functions
Create documentation for the `addXYZ` functions. See [#1833](https://github.com/ClickHouse/clickhouse-docs/issues/1833) for details on how to complete this issue. Opening issue because this is the sister issue of [#2017](https://github.com/ClickHouse/clickhouse-docs/issues/2017) (the `subtractXYZ` counterpart) - `addXYZ` was missing from the list of issues for some reason. 

- [ ] addDays
- [ ] addHours
- [ ] addInterval
- [ ] addMicroseconds
- [ ] addMilliseconds
- [ ] addMinutes
- [ ] addMonths
- [ ] addNanoseconds
- [ ] addQuarters
- [ ] addSeconds
- [ ] addTupleOfIntervals
- [ ] addWeeks
- [ ] addYears

Child issue of [#1833](https://github.com/ClickHouse/clickhouse-docs/issues/1833).

### PR DESCRIPTION
**Pull request by Blargian**
Closes [#2273](https://github.com/ClickHouse/clickhouse-docs/issues/2273) as part of the [functions project](https://github.com/ClickHouse/clickhouse-docs/issues/1833) to document all missing functions and/or update functions with insufficient documentation.

This PR either adds or updates:
-  `addDays`
-  `addHours`
-  `addInterval`
-  `addMicroseconds`
-  `addMilliseconds`
-  `addMinutes`
-  `addMonths`
-  `addNanoseconds`
-  `addQuarters`
-  `addSeconds`
-  `addTupleOfIntervals`
-  `addWeeks`
-  `addYears`

### Changelog category (leave one):
- Documentation (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit c0d2a99a9e5cb02de730fc237166c41b79d84034 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/63266/c0d2a99a9e5cb02de730fc237166c41b79d84034/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: docs/en/sql-reference/functions/date-time-functions.md [hunk 1]
```diff
@@ -2574,12 +2588,456 @@ SELECT
...
+addMilliseconds(date_time, num)
+```
+
+**Parameters**
+
+- `date_time`: Date with time to add specified number of milliseconds to. [DateTime](../data-types/datetime.md)/[DateTime64](../data-types/datetime64.md), [String](../data-types/string.md).
+- `num`: Number of milliseconds to add. [(U)Int*](../data-types/int-uint.md), [Float*](../data-types/float.md).
+
+**Returned value**
+- Returns `date_time` plus `num` milliseconds. [Date](../data-types/date.md)/[Date32](../data-types/date32.md)/[DateTime](../data-types/datetime.md)/[DateTime64](../data-types/datetime64.md).
```
Comment (rschu1ze) [inline docs/en/sql-reference/functions/date-time-functions.md:2872]: I was unable to identify a scenario / an input where `addMilliseconds` would return `Date` or `Date32`.

File: docs/en/sql-reference/functions/date-time-functions.md [hunk 2]
```diff
@@ -2574,12 +2588,456 @@ SELECT
...
+**Parameters**
+
+- `interval_1`: First interval or interval of tuples. [interval](../data-types/special-data-types/interval.md), [tuple](../data-types/tuple.md)([interval](../data-types/special-data-types/interval.md)).
+- `interval_2`: Second interval to be added. [interval](../data-types/special-data-types/interval.md).
+
+**Returned value**
+- Returns a tuple of intervals. [tuple](../data-types/tuple.md)([interval](../data-types/special-data-types/interval.md)).
+
+:::note
+If the types of the first interval (or the interval in the tuple) and the second interval are the same they will be merged into one interval.
```
Comment (rschu1ze) [inline docs/en/sql-reference/functions/date-time-functions.md:2980]: Should that be `or the intervals in the tuple`?

Comment (rschu1ze) [reply docs/en/sql-reference/functions/date-time-functions.md:2980]: More importantly, a reader won't understand what it means to "merge intervals into one intervals". Can we re-phrase a bit?

File: docs/en/sql-reference/functions/date-time-functions.md [hunk 3]
```diff
@@ -2574,12 +2588,456 @@ SELECT
...
+
+**Returned value**
+- Returns `date` with added `intervals`. [date](../data-types/date.md)/[date32](../data-types/date32.md)/[datetime](../data-types/datetime.md)/[datetime64](../data-types/datetime64.md).
+
+**Example**
+
+Query:
+
+```sql
+WITH toDate('2018-01-01') AS date SELECT addTupleOfIntervals(date, (INTERVAL 1 DAY, INTERVAL 1 MONTH, INTERVAL 1 YEAR))
```
Comment (rschu1ze) [inline docs/en/sql-reference/functions/date-time-functions.md:3030]: Please \+newline between `date` and `SELECT.

File: docs/en/sql-reference/functions/date-time-functions.md (hunk 1 above)
Comment (Blargian) [reply docs/en/sql-reference/functions/date-time-functions.md:2872]: Indeed, i see now that return type in the implementation is only `DateTime64`.

**Review by rschu1ze — APPROVED** [review]

**Pull request merged**